src/main/java/com/codeofus/rent_a_park/services/PersonService.java

```diff
@@ -0,0 +1,94 @@
+package com.codeofus.rent_a_park.services;
+
+import com.codeofus.rent_a_park.dtos.ParkingMapper;
+import com.codeofus.rent_a_park.dtos.PersonDto;
+import com.codeofus.rent_a_park.models.Person;
+import com.codeofus.rent_a_park.models.Spot;
+import com.codeofus.rent_a_park.repositories.PersonRepository;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+import java.util.Optional;
+
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+@Transactional(readOnly = true)
+@Service
+public class PersonService {
+
+    private PersonRepository personRepository;
+
+    private ParkingMapper mapper;
+
+    public List<PersonDto> getAll() {
+        return personRepository.findAll().stream().map(mapper::personToDto).toList();
+    }
+
+    @Transactional
+    public Person addNewPerson(PersonDto personDto) {
+        return personRepository.save(mapper.toPerson(personDto));
+    }
+
+    @Transactional
+    public Optional<PersonDto> updatePerson(PersonDto person) {
+        return Optional
+                .of(personRepository.findById(person.getId()))
+                .filter(Optional::isPresent)
+                .map(Optional::get)
+                .map(
+                        user -> {
+                            user.setFirstName(person.getFirstName());
+                            user.setLastName(person.getLastName());
+                            user.setRegistration(person.getRegistration());
+                            user = personRepository.save(user);
```

Comment:
Is this explicit call to personRepository save needed since we're inside of @transactional. Also, setting these values can be logic inside user class (add method void update(args){...} to User.class)